curriculum/challenges/english/25-front-end-development/lab-polygon-area-calculator/5e444147903586ffb414c94f.md

```diff
@@ -0,0 +1,881 @@
+---
+id: 5e444147903586ffb414c94f
+title: Build a Polygon Area Calculator
+challengeType: 23
+dashedName: build-a-polygon-area-calculator
+---
+
+# --description--
+
+In this project, you will use object-oriented programming to create a `Rectangle` class and a `Square` class. The `Square` class should be a subclass of `Rectangle` and inherit its methods and attributes.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should create a `rectangle` class.
+2. When a Rectangle object is created, it should be initialized with `width` and `height` attributes. The class should also contain the following methods:
+
+    - `set_width`: Sets the width of the rectangle.
+    - `set_height`: Sets the height of the rectangle.
+    - `get_area`: Returns area (`width * height`).
+    - `get_perimeter`: Returns perimeter (`2 * width + 2 * height`).
+    - `get_diagonal`: Returns diagonal (`(width ** 2 + height ** 2) ** .5`).
+    - `get_picture`: Returns a string that represents the shape using lines of '\*'. The number of lines should be equal to the height and the number of '\*' in each line should be equal to the width. There should be a new line (`\n`) at the end of each line. If the width or height is larger than `50`, this should return the string: `'Too big for picture.'`.
```

Comment:
Not sure if '*' this is supposed to be a string inside backticks or just inside backticks.
This is what it currently looks like for me

<img width="458" alt="Screenshot 2025-04-20 at 7 20 22 PM" src="https://github.com/user-attachments/assets/2ab37176-de68-44f8-a220-c8ad912b364a" />